lib/widgets/nutrition/charts.dart

```diff
@@ -120,3 +123,203 @@ class NutritionalDiaryChartWidget extends StatelessWidget {
+
+    _nutritionalPlan.logEntriesValues.forEach((key, value) {
+      if(key.difference(today).inDays >= -7) {
+        sevenDaysAvg += value;
+        count++;
+      }
+    });
+
+    if(count != 0) {
+      sevenDaysAvg.energy = sevenDaysAvg.energy/count;
+      sevenDaysAvg.protein = sevenDaysAvg.protein/count;
+      sevenDaysAvg.carbohydrates = sevenDaysAvg.carbohydrates/count;
+      sevenDaysAvg.carbohydratesSugar = sevenDaysAvg.carbohydratesSugar/count;
+      sevenDaysAvg.fat = sevenDaysAvg.fat/count;
+      sevenDaysAvg.fatSaturated = sevenDaysAvg.fatSaturated/count;
+      sevenDaysAvg.fibres = sevenDaysAvg.fibres/count;
+      sevenDaysAvg.sodium = sevenDaysAvg.sodium/count;
+    }
+
+    return sevenDaysAvg;
+  }
+
+  NutritionalValues nutritionalValuesFromPlanLogsToday() {
+    final now = DateTime.now();
+    final today = DateTime(now.year, now.month, now.day);
+
+    return _nutritionalPlan.logEntriesValues[_nutritionalPlan
+        .logEntriesValues.keys.firstWhereOrNull((d) => d.difference(today)
+        .inDays
+        == 0)]
+        ?? NutritionalValues();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final NutritionalValues loggedNutritionalValues =
+    nutritionalValuesFromPlanLogsToday();
+    final NutritionalValues sevenDayAvg =
+    nutritionalValuesFromPlanLogsSevenDayAvg();
+
+    if (_nutritionalPlan.nutritionalValues.energy == 0) {
+      return Container();
+    }
+
+    return charts.BarChart(
+
+      [
+        charts.Series<NutritionData, String>(
+          id: 'Planned',
+          domainFn: (nutritionEntry, index) => nutritionEntry.name,
+          measureFn: (nutritionEntry, index) => nutritionEntry.value,
+          data: [
+            NutritionData(
+              AppLocalizations.of(context).energy,
+              _nutritionalPlan.nutritionalValues.energy,
+            ),
+            NutritionData(
+              AppLocalizations.of(context).protein,
+              _nutritionalPlan.nutritionalValues.protein,
+            ),
+            NutritionData(
+              AppLocalizations.of(context).carbohydrates,
+              _nutritionalPlan.nutritionalValues.carbohydrates,
+            ),
+            NutritionData(
+              AppLocalizations.of(context).sugars,
+              _nutritionalPlan.nutritionalValues.carbohydratesSugar,
+            ),
+            NutritionData(
+              AppLocalizations.of(context).fat,
+              _nutritionalPlan.nutritionalValues.fat,
+            ),
+            NutritionData(
+              AppLocalizations.of(context).saturatedFat,
+              _nutritionalPlan.nutritionalValues.fatSaturated,
+            ),
+            NutritionData(
+              AppLocalizations.of(context).fibres,
+              _nutritionalPlan.nutritionalValues.fibres,
+            ),
+            NutritionData(
+              AppLocalizations.of(context).sodium,
+              _nutritionalPlan.nutritionalValues.sodium,
+            ),
+          ],
```

Comment:
Would it be very difficult to use different axis for the total engery and the other values? Right now the energy will dominate the graph:

![Screenshot_1639669067](https://user-images.githubusercontent.com/2280729/146401845-8d093560-e785-4ad0-8759-f37953ebfc41.png)

While this isn't the most realistic example, it would be cleaner and we could add a unit to the values on the axis